vaadin-grid-flow-demo/src/main/java/com/vaadin/flow/component/grid/demo/GridDemo.java

```diff
@@ -1740,11 +1742,41 @@ private void createContextSubMenu() {
             dataProvider.refreshAll();
         });
         // end-source-example
-        grid.setId("context-menu-grid");
+        grid.setId("context-sub-menu-grid");
         addCard("Context Menu", "Using Context Sub Menu With Grid", grid,
                 contextMenu);
     }
 
+    // Dynamic Context Menu begin
+    private void createDynamicContextMenu() {
+        TaskData taskData = new TaskData();
+
+        // begin-source-example
+        // source-example-heading: Dynamic Context Menu
+        Grid<Task> grid = new Grid<>();
+        ListDataProvider<Task> dataProvider = DataProvider
+                .ofCollection(taskData.getTasks());
+
+        grid.setDataProvider(dataProvider);
+        grid.addColumn(Task::getName).setHeader("Task Name");
+        grid.addColumn(Task::getDueDate).setHeader("Due Date");
+        GridContextMenu<Task> contextMenu = new GridContextMenu<>(grid);
+        contextMenu.setContextMenuDynamicContentRenderer(task -> {
+            if (task == null) {
+                // do not show the context menu when a row is not clicked
+                return false;
```

Comment:
I am not sure if that is the proper approach for handling the opening of the `contextMenu` with returning `true / false`, however I cannot propose anything better.

As discussed, probably completely ignoring the return value would be a better option.

**Not a blocker**